src/basic/path-util.c

```diff
@@ -495,6 +495,70 @@ char* path_join(const char *root, const char *path, const char *rest) {
                                rest && rest[0] == '/' ? rest+1 : rest);
 }
 
+char* path_join_many(const char *first, ...) {
+        char *joined, *q;
+        const char *p;
+        va_list ap;
+        bool slash;
+        size_t sz;
+
+        assert(first);
+
+        /* Joins all listed strings until NULL and places an "/" between them unless the strings end/begin already with
+         * one so that it is unnecessary. Note that "/" which are already duplicate won't be removed. The string
+         * returned is hence always longer than the sum of the length of each individual string.
```

Comment:
The second example in the list below is a counterexample.

Also "length**s**".